tests/suite/dataplane_perf_test.go

```diff
@@ -0,0 +1,109 @@
+package suite
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strconv"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	vegeta "github.com/tsenart/vegeta/v12/lib"
+	core "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/nginxinc/nginx-gateway-fabric/tests/framework"
+)
+
+var _ = Describe("Dataplane performance", Ordered, Label("performance"), func() {
+	files := []string{
+		"dp-perf/coffee.yaml",
+		"dp-perf/gateway.yaml",
+		"dp-perf/cafe-routes.yaml",
+	}
+	ns := &core.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "dp-perf",
+		},
+	}
+
+	var port, parentDir, resultsFilePath, addr string
+	targetURL := "http://cafe.example.com"
+	var outFile *os.File
+
+	t1 := vegeta.Target{
```

Comment:
It would be nice if the test suite didn't have to know about vegeta, in case we need to swap it out for a different load testing tool in the future. 

Can the RunLoadTest method accept a generic Target or Request object instead?